data/filter/FieldFilter.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+
+import {XH} from '@xh/hoist/core';
+import {FieldType, parseFieldValue} from '@xh/hoist/data';
+import {throwIf} from '@xh/hoist/utils/js';
+import {castArray, escapeRegExp, isArray, isEqual, isString} from 'lodash';
+
+import {Filter} from './Filter';
+
+/**
+ * Represents a filter operation on a predefined Field. Used by {@see FilterModel}.
+ * Immutable.
+ */
+export class FieldFilter extends Filter {
+
+    get isFieldFilter() {return true}
+
+    /** @member {string} */
+    field;
+    /** @member {string} */
+    operator;
+    /** @member {(*|[])} */
+    value;
+    /** @member {FieldType} */
+    fieldType;
```

Comment:
Arg it really bugs me that we need to carry the type here but I can see how the requirement to serialize / deserialize and also keep this general makes it really tricky to maintain a link to a real `Field` instance or a `Store` that could provide one from the name.  ☹️